Modules/ExternalFeed/classes/Setup/class.ilExternalFeedRemoveMigration.php

```diff
@@ -0,0 +1,218 @@
+    }
+
+    public function getDefaultAmountOfStepsPerRun() : int
+    {
+        return 10;
+    }
+
+    public function getPreconditions(Environment $environment) : array
+    {
+        return [
+            new ilIniFilesLoadedObjective(),
+            new ilDatabaseInitializedObjective(),
+            new ilDatabaseUpdatedObjective(),
+        ];
+    }
+
+    public function prepare(Environment $environment) : void
+    {
+        /** @var ilDBInterface $db */
+        $db = $environment->getResource(Environment::RESOURCE_DATABASE);
+        $this->db = $db;
+    }
+
+    protected function log(string $str) : void
+    {
+        echo "\n" . $str;
+    }
+
+    public function step(Environment $environment) : void
+    {
+        $db = $this->db;
+
+        $ref_ids = [];
+        $obj_ids = [];
+
+        $set = $db->queryF(
+            "SELECT * FROM object_data d JOIN object_reference r ON (d.obj_id = r.obj_id) " .
+            " WHERE d.type = %s ",
+            ["text"],
+            ["feed"]
+        );
+        while ($rec = $db->fetchAssoc($set)) {
+            $this->log("Found title: " . $rec["title"] . ", ref id: " . $rec["ref_id"]);
+            $ref_ids[$rec["ref_id"]] = $rec["ref_id"];
+            $obj_ids[$rec["obj_id"]] = $rec["obj_id"];
+        }
+        
+        // local roles of feed objects
+        $set = $db->queryF(
+            "SELECT * FROM rbac_fa " .
+            " WHERE " . $db->in("parent", $ref_ids, false, "integer"),
+            [],
+            []
+        );
+        $local_roles = [];
+        while ($rec = $db->fetchAssoc($set)) {
+            $local_roles[$rec["rol_id"]] = $rec["rol_id"];
+        }
+
+        // typ
+        $set = $db->queryF(
+            "SELECT * FROM object_data " .
+            " WHERE type = %s AND title = %s",
+            ["text", "text"],
+            ["typ", "feed"]
+        );
+        $typ_id = 0;
+        if ($rec = $db->fetchAssoc($set)) {
+            $typ_id = $rec["obj_id"];
+        }
+
+        // create ops id
+        $set = $db->queryF(
+            "SELECT * FROM rbac_operations " .
+            " WHERE operation = %s",
+            ["text"],
+            ["create_feed"]
+        );
+        $create_ops_id = 0;
+        if ($rec = $db->fetchAssoc($set)) {
+            $create_ops_id = $rec["ops_id"];
+        }
+
+
+        // ...tree
+        $this->log("Delete tree nodes of feed objects.");
+        $this->manipulate(
+            "DELETE FROM tree WHERE " .
+            " " . $db->in("child", $ref_ids, false, "integer")
+        );
+
+        // ...object_data (feed objects)
+        $this->log("Delete object_data entries of feed objects.");
+        $this->manipulate(
+            "DELETE FROM object_data WHERE " .
+            " " . $db->in("obj_id", $obj_ids, false, "integer")
+        );
+        // ...object_data (roles)
+        $this->log("Delete object_data entries of local roles of feed objects.");
```

Comment:
Please also delete the entries in role_data:
DELETE from role_data WHERE  $db->in("role_id", $local_roles, false, "integer")